src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -120,7 +126,16 @@
             throw new IOException("Unsupported credential type: " + credentials.getClass().getName());
         }
 
-        return gb.build();
+        GitHub github = gb.build();
+        try {
+            github.checkApiUrlValidity();
```

Comment:
But I wonder if we could not just produce a nicer message for whatever exception would result later anyway (`FileNotFoundException`?), thus avoiding the extra network call in the normal case that all is well.